photosynthesis_metrics/kid.py

```diff
@@ -1,69 +1,84 @@
+
+def compute_polynomial_mmd_averages(
+    x : np.ndarray,
+    y : np.ndarray,
+    n_subsets : int = 50,
+    subset_size : int = 1000,
+    ret_var : bool = True,
+    **kernel_args
+    ) -> Union[List[np.ndarray], Tuple[List[np.ndarray], List[np.ndarray]]]:
+    m = min(x.shape[0], y.shape[0])
     mmds = np.zeros(n_subsets)
     if ret_var:
         vars = np.zeros(n_subsets)
 
     choice = np.random.choice
 
-    with tqdm(range(n_subsets), desc='MMD', file=output) as bar:
-        for i in bar:
-            g = codes_g[choice(len(codes_g), subset_size, replace=False)]
-            r = codes_r[choice(len(codes_r), subset_size, replace=False)]
-            o = compute_polynomial_mmd(g, r, **kernel_args, var_at_m=m, ret_var=ret_var)
-            if ret_var:
-                mmds[i], vars[i] = o
-            else:
-                mmds[i] = o
-            bar.set_postfix({'mean': mmds[:i + 1].mean()})
+    for i in range(n_subsets):
+        g = x[choice(len(x), subset_size, replace=False)]
+        r = y[choice(len(y), subset_size, replace=False)]
+        o = compute_polynomial_mmd(g, r, **kernel_args, var_at_m=m, ret_var=ret_var)
+        if ret_var:
+            mmds[i], vars[i] = o
+        else:
+            mmds[i] = o
 
     return (mmds, vars) if ret_var else mmds
 
 
-def compute_polynomial_mmd(codes_g, codes_r, degree=3, gamma=None, coef0=1,
-                           var_at_m=None, ret_var=True):
+def compute_polynomial_mmd(
+    x : torch.Tensor,
+    y : torch.Tensor,
+    degree : int = 3,
+    gamma : Optional[float] = None,
+    coef0 : int = 1,
+    var_at_m : Optional[int] = None,
+    ret_var : bool = True) -> torch.Tensor:
     """
     Computes KID (polynomial MMD) for given sets of features, obtained from Inception net
     or any other feature extractor.
-
     Args:
-        codes_g(np.ndarray): 2D array of features from feature extractor e.g. (1000, 2048).
-        codes_r(np.ndarray): 2D array of features from feature extractor e.g. (1000, 2048).
-        degree(int): the distance can work with polynomial functions of any degree.
-        gamma: ---
-        coef0: ---
-        var_at_m: ---
-        ret_var(bool): whether to return variance after the distance is computed.
+        x: Samples from data distribution. Shape (N_samples, data_dim), dtype: torch.float32 in range 0 - 1.
+        y: Samples from data distribution. Shape (N_samples, data_dim), dtype: torch.float32 in range 0 - 1
+        degree: Degree of a polynomial functions used in kernels. Default: 3
+        gamma: Kernel parameter. See paper for details
+        coef0: Kernel parameter. See paper for details
+        var_at_m: Kernel variance. Default is `None`
+        ret_var: whether to return variance after the distance is computed.
                        This function will return Tuple[float, float] in this case.
-
     Returns:
         float or Tuple[float, float]: KID score and variance (optional).
```

Comment:
Return type in the docstring does not correspond to the one in the type annotation. Better just remove it
```suggestion
        KID score and variance (optional).
```